agent/src/util/discovery_operator.rs

```diff
@@ -1586,4 +1857,775 @@ pub mod tests {
             panic!("expected external stream");
         }
     }
+
+    #[tokio::test]
+    async fn test_get_discovery_properties_no_properties() {
+        let _ = env_logger::builder().is_test(true).try_init();
+        let discovery_operator = create_discovery_operator("../test/yaml/config-a.yaml");
+        let mock_kube_interface: Arc<dyn k8s::KubeInterface> = Arc::new(MockKubeInterface::new());
+
+        // properties should be empty if not specified
+        assert!(discovery_operator
+            .get_discovery_properties(mock_kube_interface, &None)
+            .await
+            .unwrap()
+            .is_empty());
+    }
+
+    #[tokio::test]
+    async fn test_get_discovery_properties_empty_property_list() {
+        let _ = env_logger::builder().is_test(true).try_init();
+        let discovery_operator = create_discovery_operator("../test/yaml/config-a.yaml");
+        let mock_kube_interface: Arc<dyn k8s::KubeInterface> = Arc::new(MockKubeInterface::new());
+        let properties = Vec::<DiscoveryProperty>::new();
+
+        // properties should be empty if property list is empty
+        assert!(discovery_operator
+            .get_discovery_properties(mock_kube_interface, &Some(properties))
+            .await
+            .unwrap()
+            .is_empty());
+    }
+
+    #[tokio::test]
+    async fn test_get_discovery_properties_value_no_value() {
+        let _ = env_logger::builder().is_test(true).try_init();
+        let discovery_operator = create_discovery_operator("../test/yaml/config-a.yaml");
+        let mock_kube_interface: Arc<dyn k8s::KubeInterface> = Arc::new(MockKubeInterface::new());
+        let property_name_1 = "property_name_1".to_string();
+        let property_name_2 = "".to_string(); // allow empty property name
+        let properties = vec![
+            DiscoveryProperty {
+                name: property_name_1.clone(),
+                value: None,
+                value_from: None,
+            },
+            DiscoveryProperty {
+                name: property_name_2.clone(),
+                value: None,
+                value_from: None,
+            },
```

Comment:
it isn't clear to me why name="", value=None, value_from=None is valid?

Reply:
we don't restrict empty name, as long as it's not None, we accept it.  For value/valueFrom, they can be None by schema definition

Reply:
Doesn't the `oneOf` section of the CRD requires that either `value` or `valueFrom` is specified ?

Reply:
yes, switch to use schema, the oneOf property makes the requirement that either value or valueFrom required.